tla/consistency/ExternalHistoryInvars.tla

```diff
@@ -205,7 +205,56 @@ CommittedRwLinearizableInv ==
     /\ AllCommittedObservedInv
     /\ AtMostOnceObservedInv
 
-\*Debugging invariants to check that specific states are reachable
+----
+
+\* Ordering over txIDs, the form of which is <<view,seqnum>>
+TxIDStrictlyLessThan(x, y) ==
+    \/ x[1] < y[1]
+    \/ /\ x[1] = y[1]
+       /\ x[2] < y[2]
+
+\* CommittedRwResponseSorted is a subset of history containing only the responses to committed rx transactions
+\* and sorted by tx_id (instead of by event ordering)
+CommittedRwResponses == 
+    SetToSortSeq({history[i]: i \in RwTxResponseCommittedEventIndexes}, 
+    LAMBDA x,y : TxIDStrictlyLessThan(x.tx_id, y.tx_id))
+
+
+\* If a transaction response is received (event i) before another transaction is requested (event j), 
+\* then tx_id of the first transaction is strictly less than the tx_id of the second transaction.
+\* Note that this invariant is only considers committed read-write transactions.
+CommittedRwOrderedRealTimeInv == 
+    \A i \in RwTxResponseCommittedEventIndexes :
+        \A j \in RwTxRequestCommittedEventIndexes :
+            \A k \in RoTxResponseCommittedEventIndexes :
+                /\ history[k].tx = history[j].tx
+                /\ i < j 
+                => TxIDStrictlyLessThan(history[i].tx_id, history[k].tx_id)
+
+\* Each transaction observes the previous transaction in the TxID order and its own write
+\* Note that this invariant is only considers committed read-write transactions.
+CommittedRwOrderedSerializableInv ==
+    \/ Len(CommittedRwResponses) < 2
+    \/ \A i \in 1..Len(CommittedRwResponses)-1:
```

Comment:
The first disjunct is redundant because `\A i \in 1..0: P` is true.

Reply:
fixed in https://github.com/microsoft/CCF/pull/6185/commits/ed2e55082fb3a27cf5f5041843af02d874b15ff5